src/cluster.py

```diff
@@ -164,11 +162,10 @@ def __init__(
         self.rewind_password = rewind_password
         self.raft_password = raft_password
         self.patroni_password = patroni_password
-        self.tls_enabled = tls_enabled
         # Variable mapping to requests library verify parameter.
         # The CA bundle file is used to validate the server certificate when
         # TLS is enabled, otherwise True is set because it's the default value.
-        self.verify = f"{PATRONI_CONF_PATH}/{TLS_CA_FILE}" if tls_enabled else True
+        self.verify = f"{PATRONI_CONF_PATH}/{TLS_CA_BUNDLE_FILE}"
```

Comment:
When switching between real peer certificate from operator and the internally generated one, the charm can get stuck validating against the wrong CA, so both are bundled inside this file to validate against in requests.

Reply:
LGTM, but can't we use "internal_CA if self.tls_enabled else relation_CA"?

Reply:
I don't think so. As is the charm property will be true if all the relation components are available (key, cert and ca), not if those are currently used by Patroni. To be able to use only one CA, we have to track which cert is being used by Patroni ATM on each unit.